scipy/stats/stats.py

```diff
@@ -8121,6 +8121,281 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
     return BrunnerMunzelResult(wbfn, p)
 
 
+def _confint_lowerbound(n, quantile, confidence):
+    r"""
+    Compute the lower bound for a one-sided confidence interval
+    for a given
+    - quantile (0<`quantile`<1)
+    - confidence level (0<`confidence`<1)
+    - number of samples `n`.
+
+    Returns the largest index of the sample being a valid lower bound,
+    or `None` if there are not enough samples to derive one.
+
+    Used by the public function confint_quantile().
+
+    .. versionadded:: 1.7.0
+    """
+
+    # compute all probabilities from the binomial distribution for
+    # the quantile of interest
+    bd = distributions.binom(n, quantile)
+
+    # the lower bound is the last index before the invert survival
+    # function value for the target confidence level
+    lb = bd.isf(confidence) - 1
+
+    if lb < 0:  # isf returns -1 if there are no matching index
+        return None
+    else:
+        return int(lb)
+
+
+def confint_quantile(x, quantile, confidence, type='one-sided'):
+    r"""Compute non-parametric confidence intervals for any quantile.
+
+    This function implements a non-parametric approach to compute confidence
+    intervals for quantiles. The approach is attributed to Thompson [1]_ and
```

Comment:
Correct me if I'm wrong, but is the word `quantile` used throughout the documentation to refer to both
1) the floating point value between 0 and 1 and 
2) the corresponding element of `x`?
If so, is it possible to disambiguate?